apple/t2/README.md

```diff
@@ -7,3 +7,32 @@ Overall, most features (WiFi, bluetooth, audio, touchbar, ...) of Macs are suppo
 Following [this guide](https://wiki.t2linux.org/distributions/nixos/installation/) is the recommended way to install, as it incudes the extra things you have to do on a T2 Mac.
 
 You can consult the [wiki](https://wiki.t2linux.org/) for information specific to T2 Macs.
+
+## Unlocking Internal iGPU
+
+The `apple-set-os-loader-installer.nix` module serves as an installer for the [`apple-set-os-loader`](https://github.com/Redecorating/apple_set_os-loader). This tool is designed to unlock the internal integrated GPU (iGPU) on certain MacBooks. See https://wiki.t2linux.org/guides/hybrid-graphics/ for more details.
+
+### What it Does:
+
+Upon activation, this module performs the following:
+
+- Renames the existing `BOOTX64.EFI` file to `bootx64_original.efi`.
+- Installs the `apple-set-os-loader` hook in its place as `bootx64.efi`.
+- Before the system boots the hook unlocks the iGPU and subsequently calls the original `bootx64_original.efi`.
+
+### How to Implement:
+
+1. **Integrate the apple-set-os-loader installer** into your `configuration.nix`:
+```
+imports = [
+  ...
+  "${builtins.fetchGit { url = "https://github.com/NixOS/nixos-hardware.git"; }}/apple/t2/apple-set-os-loader-installer.nix"
```

Comment:
That feels a bit clunky to have a separate module. If this is something we shouldn't enable by default, I would make this a nixos module option and include this in the t2 module. Than we also check that evaluation still works.

Reply:
Thanks! I'll give it a go :-)

Reply:
@Mic92 Alright, i've integrated the code into default.nix and made it optional as described in the readme.. However I had some trouble building it at first so had chatgpt helping me out tweaking a couple of things. I'm a nix noob so please let me know if i wrecked havoc :-)